// alexa-response/build/index.js
'use strict';

var alexaSsml = require('alexa-ssml');

var babelHelpers = {};
babelHelpers.typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) {
  return typeof obj;
} : function (obj) {
  return obj && typeof Symbol === "function" && obj.constructor === Symbol ? "symbol" : typeof obj;
};

babelHelpers.classCallCheck = function (instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
};

babelHelpers.createClass = function () {
  function defineProperties(target, props) {
    for (var i = 0; i < props.length; i++) {
      var descriptor = props[i];
      descriptor.enumerable = descriptor.enumerable || false;
      descriptor.configurable = true;
      if ("value" in descriptor) descriptor.writable = true;
      Object.defineProperty(target, descriptor.key, descriptor);
    }
  }

  return function (Constructor, protoProps, staticProps) {
    if (protoProps) defineProperties(Constructor.prototype, protoProps);
    if (staticProps) defineProperties(Constructor, staticProps);
    return Constructor;
  };
}();

babelHelpers.extends = Object.assign || function (target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i];

    for (var key in source) {
      if (Object.prototype.hasOwnProperty.call(source, key)) {
        target[key] = source[key];
      }
    }
  }

  return target;
};

babelHelpers.objectWithoutProperties = function (obj, keys) {
  var target = {};

  for (var i in obj) {
    if (keys.indexOf(i) >= 0) continue;
    if (!Object.prototype.hasOwnProperty.call(obj, i)) continue;
    target[i] = obj[i];
  }

  return target;
};

babelHelpers;

var CardType = {
  LinkAccount: 'LinkAccount',
  Simple: 'Simple',
  Standard: 'Standard'
};

var SpeechType = {
  PlainText: 'PlainText',
  SSML: 'SSML'
};

var Response = function () {
  function Response() {
    var state = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];
    babelHelpers.classCallCheck(this, Response);

    this.state = state;
  }

  babelHelpers.createClass(Response, [{
    key: 'ask',
    value: function ask(text, type) {
      return this.say(text, type).shouldEndSession(false);
    }
  }, {
    key: 'say',
    value: function say(text, type) {
      return new Response(babelHelpers.extends({}, this.state, {
        response: babelHelpers.extends({}, this.state.response, {
          shouldEndSession: true
        }, outputSpeech(text, type))
      }));
    }
  }, {
    key: 'reprompt',
    value: function reprompt(text, type) {
      return new Response(babelHelpers.extends({}, this.state, {
        response: babelHelpers.extends({}, this.state.response, {
          reprompt: babelHelpers.extends({}, outputSpeech(text, type))
        })
      }));
    }
  }, {
    key: 'card',
    value: function card(_ref) {
      var _ref$type = _ref.type;
      var type = _ref$type === undefined ? CardType.Simple : _ref$type;
      var rest = babelHelpers.objectWithoutProperties(_ref, ['type']);

      return new Response(babelHelpers.extends({}, this.state, {
        response: babelHelpers.extends({}, this.state.response, {
          card: babelHelpers.extends({}, rest, type && { type: type })
        })
      }));
    }
  }, {
    key: 'attributes',
    value: function attributes(data) {
      return new Response(babelHelpers.extends({}, this.state, {
        sessionAttributes: babelHelpers.extends({}, this.state.sessionAttributes, data)
      }));
    }
  }, {
    key: 'shouldEndSession',
    value: function shouldEndSession(_shouldEndSession) {
      return new Response(babelHelpers.extends({}, this.state, {
        response: babelHelpers.extends({}, this.state.response, {
          shouldEndSession: _shouldEndSession
        })
      }));
    }
  }, {
    key: 'build',
    value: function build(attributes) {
      return babelHelpers.extends({
        version: '1.0'
      }, this.state, {
        response: babelHelpers.extends({
          shouldEndSession: true
        }, this.state.response)
      }, attributes || this.state.sessionAttributes ? { sessionAttributes: babelHelpers.extends({}, attributes, this.state.sessionAttributes) } : null);
    }
  }]);
  return Response;
}();

Response.ask = function () {
  var _ref2;

  return (_ref2 = new Response()).ask.apply(_ref2, arguments);
};

Response.say = function () {
  var _ref3;

  return (_ref3 = new Response()).say.apply(_ref3, arguments);
};

Response.card = function () {
  var _ref4;

  return (_ref4 = new Response()).card.apply(_ref4, arguments);
};

Response.reprompt = function () {
  var _ref5;

  return (_ref5 = new Response()).reprompt.apply(_ref5, arguments);
};

Response.shouldEndSession = function () {
  var _ref6;

  return (_ref6 = new Response()).shouldEndSession.apply(_ref6, arguments);
};

Response.build = function (params) {
  return Object.keys(params).reduce(function (response, action) {
    var options = params[action];
    var text = typeof options === 'string' ? options : options;
    var type = typeof options === 'string' ? undefined : options.type;
    switch (action) {
      case 'ask':
        return response.ask(text, type);
      case 'say':
        return response.say(text, type);
      case 'reprompt':
        return response.reprompt(text, type);
      case 'card':
        return response.card(options);
      case 'attributes':
        return response.attributes(options);
      case 'shouldEndSession':
        return response.shouldEndSession(options);
    }
  }, new Response());
};

var outputSpeech = function outputSpeech(text) {
  var type = arguments.length <= 1 || arguments[1] === undefined ? SpeechType.PlainText : arguments[1];

  if (type === SpeechType.SSML || (typeof text === 'undefined' ? 'undefined' : babelHelpers.typeof(text)) === 'object') {
    var speech = (typeof text === 'undefined' ? 'undefined' : babelHelpers.typeof(text)) === 'object' ? alexaSsml.renderToString(text) : text;
    return { outputSpeech: { type: SpeechType.SSML, ssml: speech } };
  } else {
    return { outputSpeech: { type: type, text: text } };
  }
};

// For commonjs compatibility
var ask = Response.ask;
var say = Response.say;
var card = Response.card;
var reprompt = Response.reprompt;
var shouldEndSession = Response.shouldEndSession;
var build = Response.build;

exports.CardType = CardType;
exports.SpeechType = SpeechType;
exports['default'] = Response;
exports.ask = ask;
exports.say = say;
exports.card = card;
exports.reprompt = reprompt;
exports.shouldEndSession = shouldEndSession;
exports.build = build;
