// alexa-ssml/lib/schema/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.w = exports.speak = exports.sayas = exports.s = exports.phoneme = exports.p = exports.pause = exports.audio = undefined;

var _audio2 = require('./audio');

var _audio = _interopRequireWildcard(_audio2);

var _pause2 = require('./pause');

var _pause = _interopRequireWildcard(_pause2);

var _p2 = require('./p');

var _p = _interopRequireWildcard(_p2);

var _phoneme2 = require('./phoneme');

var _phoneme = _interopRequireWildcard(_phoneme2);

var _s2 = require('./s');

var _s = _interopRequireWildcard(_s2);

var _sayAs = require('./sayAs');

var _sayas = _interopRequireWildcard(_sayAs);

var _speak2 = require('./speak');

var _speak = _interopRequireWildcard(_speak2);

var _w2 = require('./w');

var _w = _interopRequireWildcard(_w2);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

exports.audio = _audio;
exports.pause = _pause;
exports.p = _p;
exports.phoneme = _phoneme;
exports.s = _s;
exports.sayas = _sayas;
exports.speak = _speak;
exports.w = _w;
