// alexa-ssml/lib/schema/pause.js
'use strict';

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

Object.defineProperty(exports, "__esModule", {
    value: true
});

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var tag = exports.tag = 'break';

var schema = exports.schema = {
    title: 'pause',
    description: '',

    required: [],
    type: 'object',
    properties: {
        strength: {
            enum: ['none', 'x-weak', 'weak', 'medium', 'strong', 'x-strong']
        },
        time: {
            type: 'integer',
            min: 0,
            max: 10000
        }
    }
};

var transform = exports.transform = function transform(_ref) {
    var time = _ref.time;

    var props = _objectWithoutProperties(_ref, ['time']);

    return _extends({}, props, time && { time: time + 'ms' });
};
